### DISCUSSION TITLE
findOne functions for local API

### DISCUSSION THREAD

**Discussion by tyteen4a03** [payloadcms/payload#4167] [Feature Requests & Ideas]
There are times where I only expect one entity from a `find` operation. It would be great if we can have a `findOne` function that doesn't require us to unwrap `docs`.

Additionally, it would be great to have a `findOneOrFail` function where if multiple docs match the predicate, the function throws an exception instead.

**Comment by r1tsuu** [comment]
Agree, it should also skip pagination resolving (`count` query) which actually should make it more performant than `find` with `limit: 1`

**Reply by christophemenager** [reply]
`pagination: false,` could be used to skip pagination
